src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elementData;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            elementData = growIfArrayFull();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index doesn't exist");
+        }
+        if (size >= elementData.length) {
+            elementData = growIfArrayFull();
+        }
+        //shift of positions to the right side
+        for (int i = size; i > index; i--) {
+            elementData[i] = elementData[i - 1];
+        }
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        findIndex(index);
+        return elementData[index];
+
     }
 
     @Override
     public void set(T value, int index) {
+        findIndex(index);
+        elementData[index] = value;
+
+    }
 
+    private void findIndex(int index) {
+        if (index < 0 || index > size || index >= size) {
```

Comment:
The condition `index > size || index >= size` is redundant and can be simplified. Since both conditions check if `index` is greater than or equal to `size`, you can just use `index >= size`.